kubeflow/pytorch-job/prototypes/pytorch-job.jsonnet

```diff
@@ -0,0 +1,55 @@
+// @apiVersion 0.1
+// @name io.ksonnet.pkg.pytorch-job
+// @description A PyTorch job (could be training or evaluation).
+// @shortDescription A PyTorch job.
+// @param name string Name to give to each of the components
+// @optionalParam namespace string null Namespace to use for the components. It is automatically inherited from the environment if not set.
+// @optionalParam args string null Comma separated list of arguments to pass to the job
+// @optionalParam image string null The docker image to use for the job.
+// @optionalParam image_gpu string null The docker image to use when using GPUs.
+// @optionalParam num_masters number 1 The number of masters to use
+// @optionalParam num_ps number 0 The number of ps to use
+// @optionalParam num_workers number 0 The number of workers to use
+// @optionalParam num_gpus number 0 The number of GPUs to attach to workers.
+
+// TODO(https://github.com/ksonnet/ksonnet/issues/235): ks param set args won't work if the arg starts with "--".
+
+local k = import "k.libsonnet";
```

Comment:
I suggest putting all all logic into the library and just passing to the library params and env.
Example
https://github.com/kubeflow/kubeflow/blob/master/kubeflow/new-package-stub/prototypes/tf-prototype.jsonnet

This should be easier to maintain.